IBX-4981 Page Builder for B2B Portals (#1971)
ezsystems/developer-documentation: docs/customer_management/cp_page_builder.md

```yaml
ibexa:
    system:
        page_builder:
            siteaccess_list:
                - site
                - corporate
                - custom_portal
```

## Customer Portal container

Now, you can go back to your **Customer Portals** root folder and select **Create content**.
There you will see two possibilities **Customer Portal** and **Customer Portal Page**.

The first one is a separate Content Type used as a container for your Customer Portal pages.
Customer Portals containers should be used to sort Customer Portal pages and any other Content Types used by them, such as articles, inside the root folder.
We highly recommend that you use them instead of folders to divide and store your portals.

![Create content tab](img/cp_portal_vs_page.png)

## Assigning Portal to Customer Group

You can assign multiple Customer Portal containers or Pages to a specific Customer Group.
First, you need to grant the following permissions to company members from the Customer Group: 

- `user/login` to `custom_portal` SiteAccess
- `content/read` to selected Customer Portals

![Customer Portal permissions](img/cp_permissions.png)

> Next, add portal mapping configuration in `config/services.yaml`:

Reviewer: That part is only important/required if you are planning to separate your portals by customer groups, it is not mandatory otherwise.

What is missing here is that you can specify your own custom logic for redirecting members to specific portals by implementing `\Ibexa\Contracts\CorporateAccount\CustomerPortal\PickRule\CustomerPortalPickRule` and taggin it with `ibexa.corporate_account.customer_portal.pick_rule`

---
description: Create unique Customer Portals for your clients with Page Builder.
edition: experience
---

# Create Customer Portal

On this page, you will learn how to configure the Customer Portal feature to be editable with Page Builder.
If you already configured Customer Portal, you can learn how to build it with a Page Builder in [User Documentation.]([[= user_doc =]]/customer_management/build_customer_portal/) 

## Configure Page Builder access to Customer Portal

> First, you need to create a root folder for Customer Portals, 

Reviewer: Since https://github.com/ibexa/corporate-account/pull/175 it is also possible to mark `Customer Portal` CT as root, then only that one portal is accessible for given site access. But this is more like special case, not sure if it should be in that paragraph.

its `location_id` will be later specified in the configuration as [a tree root](multisite_configuration.md#location-tree).

> To do it, go to **Content**->**Content structure** and add a new folder in the Content tree where you will be adding Customer Portals.

Reviewer 2: ```suggestion
To do it, go to **Content** -> **Content structure**, and create folder in the Content tree where you will add Customer Portals.
```

![Customer Portals folder](img/cp_folder_for_portals.png)

To be able to see Customer Portal site in the Page Builder you need to add `custom_portal` SiteAccess to the configuration.
First, go to `config/packages/ibexa.yaml` and add `custom_portal` to
the SiteAccess `list` and to `corporate_group`.

> Next add configuration for `corporate_group` and `custom_portal` under `system`.

Reviewer 2: ```suggestion
Next, add configuration for `corporate_group` and `custom_portal` under `system`.
```

> Remember to specify `location_id` of the root folder for Customer Portals, you will find it under **Details**. 

Reviewer 2: ```suggestion
Remember to specify `location_id` of the root folder for Customer Portals, you can find it under the **Details** tab. 
```

        corporate_group:
            languages: [eng-GB]
        custom_portal:
            languages: [ eng-GB ]
            content:
                tree_root:
                    location_id: location_id_of_customer_portals_root_folder
                    excluded_uri_prefixes: [ /media, /images ]
```

Next, go to `config/packages/ibexa_admin_ui.yaml` and add `custom_portal` to the SiteAccess list available to Page Builder:

```yaml
ibexa:
    system:
        page_builder:
            siteaccess_list:
                - site
                - corporate
                - custom_portal
```

## Customer Portal container

Now, you can go back to your **Customer Portals** root folder and select **Create content**.
There you will see two possibilities **Customer Portal** and **Customer Portal Page**.

The first one is a separate Content Type used as a container for your Customer Portal pages.
Customer Portals containers should be used to sort Customer Portal pages and any other Content Types used by them, such as articles, inside the root folder.
It is recommended that you use them instead of folders to divide and store your portals.

![Create content tab](img/cp_portal_vs_page.png)

> ## Assigning Portal to Customer Group

Reviewer 2: ```suggestion
## Assign Portal to Customer Group
```

You can assign multiple Customer Portal containers or Pages to a specific Customer Group.
First, you need to grant the following permissions to company members from the Customer Group: 

- `user/login` to `custom_portal` SiteAccess
- `content/read` to selected Customer Portals

![Customer Portal permissions](img/cp_permissions.png)

> If members of the Customer Group don't have sufficient permissions for any Customer Portal assigned to them, they will be transferred to the default Customer Portal view.

Reviewer 2: ```suggestion
If members of the Customer Group don't have sufficient permissions for any Customer Portal assigned to them, they will be redirected to the default Customer Portal view.
```

!!! note

>     Customer Portal is only available to users that are members of the company. Even if user has all the sufficient permissions but is not a meber of a company, they will not see the Customer Portal.

Reviewer 2: ```suggestion
    Customer Portal is only available to users that are members of the company. Even if user has all the sufficient permissions but is not a member of a company, they won't see the Customer Portal.
```

### Build-in portal mapping

> The next step is only required if you are planning to separate your portals by customer groups.

Reviewer 2: ```suggestion
The next step is only required if you plan to separate your portals by customer groups.
```

> If you have no intention of doing that, you can skip it and move on to [Change Customer Portal layout.](#change-customer-portal-layout)

Reviewer 2: ```suggestion
If you don't plan it, skip this step and move on to [Change Customer Portal layout.](#change-customer-portal-layout)
```
